data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/HasTagsExpressionFunction.java

```diff
@@ -21,11 +21,14 @@ public Object evaluate(final List<Object> args, Event event, Function<Object, Ob
         if (args.size() == 0) {
             throw new RuntimeException("hasTags() takes at least one argument");
         }
-        if(!args.stream().allMatch(a -> a instanceof String)) {
-            throw new RuntimeException("hasTags() takes only String type arguments");
+        if(!args.stream().allMatch(a -> ((a instanceof String) && (((String)a).length() > 0) && (((String)a).charAt(0) == '"')))) {
```

Comment:
I'd probably consider splitting this so that if this error pops up, we can better know which situation it is. But, I don't think this is critical for now.